src/Microsoft.Health.Fhir.Core/Features/Search/SearchService.cs

```diff
@@ -56,6 +56,17 @@ protected SearchService(ISearchOptionsFactory searchOptionsFactory, IBundleFacto
             return _bundleFactory.CreateSearchBundle(resourceType, searchOptions.UnsupportedSearchParams, result);
         }
 
+        /// <inheritdoc />
```

Comment:
Do we need new method, handler and Request/Response? Maybe it's time to refactor the code so that search will simply take SearchOptions and move the SearchOptionsFactory to layer above and simplify the code.

Reply:
imho, it is cleaner this way, having compartment search its own handlers. Opened an issue to consider this refactor. https://github.com/Microsoft/fhir-server/issues/201.